### PR TITLE
Feature/improve output

### LINKED ISSUE #2695: Issues with the JSON created by conan create
Conan version: 1.2.0

Command: `conan create --json ./foo.json . myteam/unstable`

The generated JSON does not separate the dependency packages of the built project, from the built package. This makes it difficult to parse.

### PR DESCRIPTION
**Pull request by lasote**
- [x] Refer to the issue that supports this Pull Request. Closes #2695 
- [x] If the issue has missing info, explain the purpose/use case/pain/need that covers this Pull Request.
- [x] I've read the [Contributing guide](https://raw.githubusercontent.com/conan-io/conan/develop/.github/CONTRIBUTING.md).
- [x] I've followed the PEP8 style guides for Python code.
- [x] I've opened another PR in the Conan docs repo to the ``develop`` branch, documenting this one. Also adding a description of the changes in the ``changelog.rst`` file. https://github.com/conan-io/doc

https://github.com/conan-io/docs/pull/621

### CODE REVIEW

File: conans/client/command.py [hunk 1]
```diff
@@ -216,17 +216,21 @@ def create(self, *args):
...
+                                      args.manifests, args.manifests_interactive,
+                                      args.remote, args.update,
+                                      test_build_folder=args.test_build_folder)
+        except ConanException as exc:
+            info = exc.info
+            raise
         finally:
             if args.json:
-                self._outputer.json_install(self._conan.recorder.get_install_info(), args.json, cwd)
+                self._outputer.json_install(info, args.json, cwd)
```
Comment (memsharded) [inline conans/client/command.py]: Will raise ``info not defined`` for non ConanException

File: conans/client/conan_api.py [hunk 2]
```diff
@@ -197,12 +203,21 @@ def __init__(self, client_cache, user_io, runner, remote_manager, search_manager
...
-                                     _settings_preprocessor, self.recorder)
+        self._search_manager = search_manager
+        self._settings_preprocessor = _settings_preprocessor
+        self._recorder = None
+        self._manager = None
+
         if not interactive:
             self._user_io.disable_input()
 
+    def init_manager(self):
```
Comment (memsharded) [inline conans/client/conan_api.py]: Probably want to make this private, it is not part of the public API right?

File: conans/test/model/transitive_reqs_test.py [hunk 3]
```diff
@@ -3,6 +3,7 @@
 
 from collections import namedtuple
 
+from conans.client.action_recorder import ActionRecorder
```
Comment (memsharded) [inline conans/test/model/transitive_reqs_test.py]: Unused?

File: conans/test/model/version_ranges_test.py [hunk 4]
```diff
@@ -1,4 +1,6 @@
 import unittest
+
+from conans.client.action_recorder import ActionRecorder
```
Comment (memsharded) [inline conans/test/model/version_ranges_test.py]: Unused?

File: conans/errors.py [hunk 5]
```diff
@@ -68,7 +68,9 @@ class ConanException(Exception):
     """
          Generic conans exception
     """
-    pass
+    def __init__(self, *args, **kwargs):
+        self.info = None
```
Comment (memsharded) [inline conans/errors.py]: No one is setting the info yet?

Comment (lasote) [reply conans/errors.py]: Not currently, only adjusted in already created exceptions, so I prefer to keep it as it is to avoid confusions.

**Comment by lasote** [comment]
Pushed again, please check

**Pull request merged**